packages/utils/src/error-handlers.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * @module ErrorHandlers
+ * This module is designed to provide utility functions for validating
+ * function parameters. It includes functions that throw type errors if
+ * the parameters do not meet specified criteria, such as being defined,
+ * a number, a string, a function, or an array. This module helps ensure
+ * that functions receive the correct types of inputs, enhancing code
+ * reliability and reducing runtime errors.
+ */
+
+import {
+    SupportedType,
+    isArray,
+    isBigInt,
+    isBigNumberish,
+    isDefined,
+    isFunction,
+    isHexadecimal,
+    isNumber,
+    isObject,
+    isString,
+    isStringifiedBigInt,
+    isSupportedType,
+    isType,
+    isUint8Array
+} from "./type-checks"
+
+/**
+ * It throws a type error if the parameter value has not been defined.
+ * @param parameterValue The parameter value.
+ * @param parameterName The parameter name.
+ */
+export function requireDefined(parameterValue: any, parameterName: string) {
+    if (!isDefined(parameterValue)) {
+        throw new TypeError(`Parameter '${parameterName}' is not defined`)
```

Comment:
Including the value might be tricky because it could be an object without any `.toString()` method, or an array, or any type that would make the message hard to read.

I've added the type to the error message where the check is on a native type. Having:

`Parameter '${parameterName}' is not a stringified bigint, but received type: string` could not be very precise 🤔